dropwizard-jdbi3/src/main/java/io/dropwizard/jdbi3/JdbiHealthCheck.java

```diff
@@ -0,0 +1,35 @@
+package io.dropwizard.jdbi3;
+
+import com.codahale.metrics.health.HealthCheck;
+import com.google.common.util.concurrent.MoreExecutors;
+import io.dropwizard.db.TimeBoundHealthCheck;
+import io.dropwizard.util.Duration;
+import org.jdbi.v3.core.Jdbi;
+
+import java.util.concurrent.ExecutorService;
+
+public class JdbiHealthCheck extends HealthCheck {
+    private final Jdbi jdbi;
+    private final String validationQuery;
+    private final TimeBoundHealthCheck timeBoundHealthCheck;
+
+    public JdbiHealthCheck(ExecutorService executorService, Duration duration, Jdbi jdbi, String validationQuery) {
+        this.jdbi = jdbi;
+        this.validationQuery = validationQuery;
+        this.timeBoundHealthCheck = new TimeBoundHealthCheck(executorService, duration);
+    }
+
+    public JdbiHealthCheck(Jdbi jdbi, String validationQuery) {
+        this(MoreExecutors.newDirectExecutorService(), Duration.seconds(0), jdbi, validationQuery);
+    }
+
+    @Override
+    protected Result check() throws Exception {
+        return timeBoundHealthCheck.check(() ->
+            jdbi.withHandle((handle) -> {
```

Comment:
On second thought, I think it would be better to use ` try (Handle handle = jdbi.open())` instead of `withHandle`. The biggest advantages of lambda-expressions is that we could use a short-form for one-line calls or even reference a method. Here we use two commands, so basically the both expressions require the same amount of code. At the same time the former version is easier to test.